UXPin/uxpin-merge-tools
# Upgrade Webpack latest version (5.88.1)

## Goal

Even if Webpack is now a peer-dependency (meaning that users have to bring their own Webpack), when we run commands locally using `yarn link`, we run the version installed with the `devDependencies` of this CLI, currently `5.28.0`.

We noticed errors when upgrading from Node.js 16 to 18 so it's important to ensure we support correctly the latest version of Webpack (`5.88`)

#### Error with Node.js 18 and Webpack 5.28

```
error:0308010C:digital envelope routines::unsupported
```

![image](https://github.com/UXPin/uxpin-merge-tools/assets/5546996/30448100-3ee7-4da8-b26d-73e473f581cf)

## How to check

- Check out the branch, run `yarn install` and `make build`
- Find a Merge integration repo linked with Merge CLI, using `yarn link @uxpin/merge-cli`
- Switch to Node.js 18 with `nvm`
- Launch the experimental mode in `debug` mode to check the version of Webpack that actually runs: `DEBUG=uxpin* yarn uxpin --skip-browser` for example
- There should be no error!
 
![image](https://github.com/UXPin/uxpin-merge-tools/assets/5546996/50307669-783f-4d1d-b178-f35015975043)
